### DISCUSSION TITLE
Problem sending with dbus system

### DISCUSSION THREAD

**Discussion by Phil0x4a** [AsamK/signal-cli#1451] [Q&A] [answered]
I'm using signal-cli as a system service on dbus. This worked well for years now. After applying the update 0.12.8 the system bus shows a strange behavior, while I don't believe it is the specific version which only changed the user-agent string. 

The service is running and shows no errors. 

Using the command below delivers the message as expected
$: signal-cli --dbus-system send -m "Hello 123" +49123456789
1707211276967

Using python with dbus and the following code: 
...
globSignal = bus.get('org.asamk.Signal')
globSignal.sendMessage("Hello 123", [], "+49123456789")
1707239642239
...

It appears to work fine and also returns the timestamp. However, the message is never delivered to the recipient. It seems to be a blackhole. Also the signal-cli daemon shows no logging on the python way. Nothing happens. 

Any idea what the problem could be here?

**Comment by gitpower2017** [comment]
I have the same problem since 0.12.7.
However, this is a bit crazier.
A small test script works. The same in the main programme causes the same errors.
Everything looks fine, but the message does not arrive. Black hole fits exactly.
I now have a workaround (os.system) and use signal-cli on the console.

**Comment by Phil0x4a** [comment]
@gitpower2017: Good advise. I did the same and used os.system as a workaround. But instead of signal-cli I used dbus-send which avoids loading of jvm and should be much faster. This for now works very well for me. 

See this documentation: https://github.com/AsamK/signal-cli/wiki/DBus-service#send-using-dbus-send

**Comment by aapeliv** [comment]
Same issue, upgraded from v0.11.11 to v0.12.8 and messages are not being received (but read receipts are). No errors, just blackholed. Running signal-cli (native linux) in daemon mode on latest Arch docker container.

**Comment by AsamK** [comment, marked as answer]
Python doesn't seem to handle the overloaded sendMessage method well and occasionally selects the wrong one.
` globSignal.sendMessage("Hello 123", [], ["+49123456789"])` might work.

**Reply by aapeliv** [reply]
I have this, which used to work:

```py
def send_message(message, recipient):
    try:
        return signal.sendMessage(message, [], recipient)
    except TypeError:
        return signal.sendMessage(message, [], [recipient])
    raise Exception("Should get here")
```

**Reply by Phil0x4a** [reply]
@aapeliv: This will probably not work as expected, as the first function call doesn't throw an exception if the message is not being delivered. It just calls the wrong method, as AsamK outlined. You should use the call with the string array. 
```
def send_message(message, recipient):
    try:
        return signal.sendMessage(message, [], recipient)
    except TypeError:
        return signal.sendMessage(message, [], [recipient])
    raise Exception("Should get here")
```

**Comment by Phil0x4a** [comment]
@AsamK: You're right. Using the array instead of the single string solves it. Thank you for your help. Much appreciated. 

Solution: 
`globSignal.sendMessage("Hello 123", [], ["+49123456789"])`

**Comment by greenpines65** [comment]
Hello @philsunshine @aapeliv @gitpower2017 
The problem is described here: https://github.com/AsamK/signal-cli/discussions/1302#discussioncomment-9104254

Basically, what you see is actually several attempts to send your message to invalid numbers.
Please be careful and check that you haven't sent any sensitive information to unwanted recipients!
